### DISCUSSION TITLE
Jetbrains Copilot Chat multiple line input

### DISCUSSION THREAD

**Discussion by dmwyatt** [community/community#80395] [Copilot Conversations]
### Select Topic Area

Bug

### Body

You can no longer use Shift-Enter to add a newline to a chat.

IDE: PyCharm
Copilot plugin version: 1.4.5.4049

When you press Shift-Enter it adds like 50 blank lines instead of just one.

**Comment by arcticlinux** [comment]
Having the same issue.  I think the actual number of lines is 130 though.

Also mentioned in https://github.com/orgs/community/discussions/69250

**Reply by zzJinux** [reply]
Link to the mention: https://github.com/orgs/community/discussions/69250#discussioncomment-7988175

**Comment by eneumann** [comment]
Having the same problem but even worse, PyCharm goes unresponsive for a minute or two before adding all the blank lines. Makes it completely un-usable.

**Reply by eneumann** [reply]
Latest copilot update in PyCharm fixed this for me! 🎉

**Comment by bjork-dev** [comment]
In webstorm this maneuver adds 661 blank lines 👌

**Comment by guaycuru** [comment]
Also happening in Idea Ultimate.

**Comment by L-Sypniewski** [comment]
The same happens in Rider, sometimes it becomes unresponsive for a while - both on macOS and Ubuntu.

**Comment by evelynn996** [comment]
Copilot performs best in VSCode, where Alt + Enter functions correctly for line breaks. However, in WebStorm and IntelliJ IDEA, it adds numerous blank lines and leads to one to two minutes of unresponsiveness.

**Comment by zzJinux** [comment]
My case: https://github.com/orgs/community/discussions/84504

**Comment by LiamKarlMitchell** [comment]
Yikes this is horrible, is there any update on a fix for this bug?
Happening in Phpstorm on Windows.

**Comment by DavidHazzard** [comment]
Shift+Enter for me is opening up the Data Sources and Drivers window. Any update on this?

I'm running PyCharm on Windows with the latest copilot update.

**Reply by LiamKarlMitchell** [reply]
It did that for me in Rust Rover too you can remove the keybind in settings.

**Reply by DavidHazzard** [reply]
Ah gotcha. It wasn't doing that until I updated PyCharm. I guess some settings got reset. Thanks!

**Reply by NingAnMe** [reply]
search "source" in Keymap.

**Reply by MannyDiera** [reply]
This is it, thanks!

**Comment by angenue** [comment]
i cant believe they havent fixed this

**Comment by Harin3Bone** [comment]
Any update on this ?

**Reply by L-Sypniewski** [reply]
For Rider it's been fixed some time ago, have you tried if it works for you with updated extension?

**Reply by Harin3Bone** [reply]
Thank you for mention me, I've update only Github copilot plugin but it didn't work.
So I update Intellij Ultimate and Github Copilot. Now it's working fine.

**Comment by Travis-Barton** [comment]
Any update on this? newlines make the text input experience much nicer

**Comment by Varin6** [comment]
Currently in Rider 2024.2.4 Copilot plugin does not add new line at all on Shift+Enter. It used to in previous versions, now I am forced to write everything in single line as well as pasted code examples

**Reply by tkoenig89** [reply]
This happens for me as well. Until i found this post, i thought i had messed up my keyboard shortcuts and was search like crazy to fix it. Not having access to line breaks when writing prompts feels wrong. Please fix this.

**Reply by tkoenig89** [reply]
What is really strange: I just tried installing older versions of the Copilot plugin and went back to a version from July 2024. But Shift+Enter did not work. I would bet, that it did work back then. So maybe the problem is with the newer version of Rider.

Update: Shift+Enter is working with the latest Copilot Plugin in Rider 2024.1.6 or 2024.2.1.
Opened an issue/discussion here: https://rider-support.jetbrains.com/hc/en-us/community/posts/21416159783954-Rider-2024-2-4-Copilot-Plugin-Shift-Enter-not-working

**Comment by DanOpcode** [comment]
It used to work in Rider, but not anymore for me since some time back. I thought it would get noticed and fixed, therefore I haven't cared to search for a solution. But now it has been broken for quite some time.

**Reply by glen-fdce** [reply]
Another plugin update this morning (1.5.24) and this still isn't fixed. Assuming it's not being tested in Rider.

Note this issue affects Rider on both Windows and Mac.

**Reply by tkoenig89** [reply]
As older versions of Rider or the ( latest ) versions of other IDEs I use (like Webstorm and PyCharm) do not suffer from the same issue, i find it hard to blame this on the plugin alone. But i might be wrong on this.

**Comment by tkoenig89** [comment]
So for me i found there is one keybinding in the latest version of Rider which seems to break Shift-Enter newlines in Copilot Chat.

![image](https://github.com/user-attachments/assets/84bd337c-1233-4bce-8bda-a621153d98c1)

I never set or changed this in the past, and it was never an issue until Rider above 2024.2.1. But when i remove this keybinding i instantly can add newlines in my Copilot Chat window. 

So maybe this helps some of you having the same issue.

**Reply by Varin6** [reply]
Yup, that fixed it for me! Also not had this keybinding before. Removed it and now it works fine in Copilot Chat!

**Reply by DanOpcode** [reply]
Wow, thank you. Great fix/workaround! Once again I feel like a human using GitHub Copilot Chat!

**Reply by glen-fdce** [reply]
Thank you!!!

**Reply by Varin6** [reply]
@tkoenig89 BTW, does your 'delete' keyboard key work in Copilot Chat window?

**Reply by tkoenig89** [reply]
@Varin6 yes, delete is working without any issue

**Reply by fakkio** [reply]
> @tkoenig89 BTW, does your 'delete' keyboard key work in Copilot Chat window?

My delete was also broken, i removed the "delete" shortcut for "Remove selected tests (Tool window)" and it now works.

**Reply by KevThompson2020** [reply]
Thanks,  fixed it for me.

**Reply by Travis-Barton** [reply]
~~This does not work for me unfortunately /:~~

Nvm it did work, but i had to remove ALL keybindings related to shift + enter, which included jupyter notebook cell commands... which kinda sucks

**Comment by suchakreepanyawong** [comment]
Response #1

**Comment by ChadAPSheridan** [comment]
PhpStorm has entered the chat...
As of the 2024.3 update released 11/13/2024 this problem as spread to phpStorm as well.  Not really a fan of deleting keybinds for other plugins/tools though.

I've made a suggestion for a quick fix to the copilot plugin itself that hopefully someone on the dev team will read:
[Discussion: 110614](https://github.com/orgs/community/discussions/110614?sort=new#discussioncomment-11309414)

**Reply by jiec-msft** [reply]
@ChadAPSheridan Thank you for the advice! The shift-enter is impact not only to Copilot plugin. And looks like this issue is resolved by JetBrains in [link](https://youtrack.jetbrains.com/issue/DBE-22637/Windows-ShiftEnter-opens-Data-Sources-and-Drivers-window-when-focused-in-any-window-or-toolwindow) now since following JetBrains build versions. I tried their EAP version `251.21418.42`, and new line is back. Could you take a look next time when you upgrade to latest JetBrains IDE?
![image](https://github.com/user-attachments/assets/f9a50f25-1c92-4ca9-8a0a-447c1069692d)

**Comment by oleg-shatin-ptech** [comment]
The main reason why I'm avoiding updates: you never know, what part of IDE experience fall off next hhh
Updated to latest Rider and Shift+Enter stopped working
Rider 2024.3 Build #RD-243.21565.191 + Copilot v1.5.29.7524

**Reply by jiec-msft** [reply]
@oleg-shatin-ptech Looks like this issue is resolved by JetBrains in [link](https://youtrack.jetbrains.com/issue/DBE-22637/Windows-ShiftEnter-opens-Data-Sources-and-Drivers-window-when-focused-in-any-window-or-toolwindow) now since following JetBrains build versions. I tried their EAP version `251.21418.42`, and new line is back. Could you have a try with their latest IDE version?
![image](https://github.com/user-attachments/assets/f9a50f25-1c92-4ca9-8a0a-447c1069692d)

**Comment by seahawks8** [comment]
same thing happened to me after installing rider, all my other IDEs (rust, pycharm, etc) were impacted.
This solved it for me in keymaps (annoying but whatever)
**Remove the "Shift+Enter on the Data Source Properties**
![image](https://github.com/user-attachments/assets/ab29e626-8f72-4b89-840b-30a161ed777e)

**Reply by cdwieber** [reply]
This is exactly what fixed it for me, still appears to be an issue.

**Reply by jiec-msft** [reply]
Looks like this issue is resolved by JetBrains in [link](https://youtrack.jetbrains.com/issue/DBE-22637/Windows-ShiftEnter-opens-Data-Sources-and-Drivers-window-when-focused-in-any-window-or-toolwindow) now since following JetBrains build versions. I tried their EAP version `251.21418.42`, and new line is back. 
You may have a try when you upgraded to latest JetBrains IDE next time.
![image](https://github.com/user-attachments/assets/f9a50f25-1c92-4ca9-8a0a-447c1069692d)

**Comment by chrisyarbrough** [comment]
I would expect the fix to this issue to be: Copilot Chat should register its keyboard shortcut Shift + Enter as a "real shortcut" with the IDE, and not bypass the shortcut manager. This way, Rider would show a conflict warning and let users resolve the issue, and also it would support accessibility best-practices.

**Reply by jiec-msft** [reply]
@chrisyarbrough Looks like this issue is resolved by JetBrains in [link](https://youtrack.jetbrains.com/issue/DBE-22637/Windows-ShiftEnter-opens-Data-Sources-and-Drivers-window-when-focused-in-any-window-or-toolwindow) now since following JetBrains build versions. I tried their EAP version `251.21418.42`, and new line is back. Could you have a try with their latest IDE version?
![image](https://github.com/user-attachments/assets/f9a50f25-1c92-4ca9-8a0a-447c1069692d)
